opslevel/resource_opslevel_property_assignment.go

```diff
@@ -1,113 +1,174 @@
-// 	}
-// 	if !opslevel.IsID(definitionId) {
-// 		return fmt.Errorf("[%s] invalid definitionId", definitionId)
-// 	}
-
-// 	resource, err := client.GetProperty(ownerId, definitionId)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	// if resource was fetched correctly, attach the id to the resource
-// 	d.SetId(fmt.Sprintf("%s:%s", ownerId, definitionId))
-
-// 	if err := d.Set("definition", d.Get("definition")); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("owner", d.Get("owner")); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("value", string(*resource.Value)); err != nil {
-// 		return err
-// 	}
-
-// 	return nil
-// }
-
-// func resourcePropertyAssignmentDelete(d *schema.ResourceData, client *opslevel.Client) error {
-// 	id := strings.Split(d.Id(), ":")
-// 	if len(id) != 2 {
-// 		return fmt.Errorf("[%s] invalid property assignment id, should be in format 'ownerId:definitionId' (only a single colon between both ids, no spaces or special characters)", d.Id())
-// 	}
-// 	ownerId := id[0]
-// 	definitionId := id[1]
-// 	err := client.PropertyUnassign(ownerId, definitionId)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.SetId("")
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &PropertyAssignmentResource{}
+
+type PropertyAssignmentResource struct {
+	CommonResourceClient
+}
+
+func NewPropertyAssignmentResource() resource.Resource {
+	return &PropertyAssignmentResource{}
+}
+
+type PropertyAssignmentResourceModel struct {
+	Definition  types.String `tfsdk:"definition"`
+	Id          types.String `tfsdk:"id"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+	Locked      types.Bool   `tfsdk:"locked"`
+	Owner       types.String `tfsdk:"owner"`
+	Value       types.String `tfsdk:"value"`
+}
+
+func NewPropertyAssignmentResourceModel(assignment opslevel.Property) PropertyAssignmentResourceModel {
+	model := PropertyAssignmentResourceModel{
+		Locked: types.BoolValue(assignment.Locked),
+		Value:  types.StringValue(string(*assignment.Value)),
+	}
+	// TODO: do we need to keep using this method of setting an ID in the new plugin version?
+	// the API does not have unique ID's for property assignments, so what we did in the past was use <owner_id>:<definition_id>
+	// keeping this for now just in case it's necessary for backwards compatability.
+	model.Id = RequiredStringValue(fmt.Sprintf("%s:%s", assignment.Owner.Id(), assignment.Definition.Id))
```

Comment:
I don't think ID does anything in the new version, but I am keeping it just in case it's required for people migrating onto the new version.

Reply:
Added this to kitchen sink: https://github.com/OpsLevel/team-platform/issues/315